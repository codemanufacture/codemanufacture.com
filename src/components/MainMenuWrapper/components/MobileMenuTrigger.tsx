import * as React from 'react'
import styled from 'styled-components'
import { colors, displayDimensions } from '../../../theme'

const StyledMobileMenuTrigger = styled.a`
  display: none;
  padding: 24px 22px;
  position: absolute;
  right: 0;
  color: ${colors.link};
  font-size: 12px;
  text-decoration: none;
  text-transform: uppercase;
  letter-spacing: 3px;

  @media (max-width: ${displayDimensions.tabletSize}) {
    display: block;
  }
`

interface MobileMenuTriggerProps {
  onClick: (e: React.MouseEvent) => void
}

const MobileMenuTrigger: React.FunctionComponent<
  MobileMenuTriggerProps
> = props => (
  <StyledMobileMenuTrigger
    href="#trigger-mobile-menu"
    onClick={e => props.onClick(e)}
  >
    Menu
  </StyledMobileMenuTrigger>
)

export default MobileMenuTrigger
