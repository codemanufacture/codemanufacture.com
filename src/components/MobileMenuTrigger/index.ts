import MobileMenuTrigger from './MobileMenuTrigger'

export default MobileMenuTrigger
