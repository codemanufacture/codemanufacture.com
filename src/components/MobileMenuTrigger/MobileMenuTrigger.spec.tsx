import * as React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import MobileMenuTrigger from './MobileMenuTrigger'

describe('MobileMenuTrigger component', () => {
  it('renders correctly', () => {
    const props = {
      onClick: jest.fn(),
    }

    const tree = renderer.create(<MobileMenuTrigger {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
