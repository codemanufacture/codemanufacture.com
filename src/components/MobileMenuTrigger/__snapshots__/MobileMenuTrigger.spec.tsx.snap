// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileMenuTrigger component renders correctly 1`] = `
<a
  className="sc-bdVaJa jdAqut"
  href="#trigger-mobile-menu"
  onClick={[Function]}
>
  Menu
</a>
`;
